src/django_keycloak/middleware.py

```diff
@@ -41,14 +42,18 @@ def get_token_from_request(self, request) -> Optional[Token]:
             # Try to build a Token instance from decoded credentials
             token = Token.from_credentials(decoded_username, decoded_password)
             if token:
-                # Convert the request "Basic" auth to "Bearer" with access token
-                request.META[AUTH_HEADER] = f"Bearer {token.access_token}"
+                # Convert the request "Basic" auth to token-based with access token
+                request.META[
+                    AUTH_HEADER
+                ] = f"{settings.TOKEN_PREFIX} {token.access_token}"
             else:
-                # Setup an invalid dummy bearer token
-                request.META[AUTH_HEADER] = "Bearer not-valid-token"
+                # Setup an invalid dummy token
+                request.META[AUTH_HEADER] = f"{settings.TOKEN_PREFIX} not-valid-token"
```

Comment:
The MR looks good. Just wanted to ask why an invalid token is saved to the auth header

Reply:
This was already implemented. The current code:
```python
# Setup an invalid dummy bearer token
request.META[AUTH_HEADER] = "Bearer not-valid-token"
```
I only changed `Bearer` to match the custom `TOKEN_PREFIX`. I'm not 100% sure, but I believe that it is because this code runs when user used basic auth but they are invalid. So when we doing automatic casting to token-based auth we setup an invalid token as well to have 401 error instead of deleting credentials and having 403.

Reply:
Ah OK. It would interfere with authentication if a user has multiple auth middlewares but until that requirement pops up, lets just resolve it and keep an eye on this bit of code.